staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factory.go

```diff
@@ -105,9 +106,10 @@ type sharedInformerFactory struct {
 }
 
 // NewSharedInformerFactory constructs a new instance of sharedInformerFactory
-func NewSharedInformerFactory(client {{.clientSetInterface|raw}}, defaultResync {{.timeDuration|raw}}) SharedInformerFactory {
+func NewSharedInformerFactory(client {{.clientSetInterface|raw}}, namespace string, defaultResync {{.timeDuration|raw}}) SharedInformerFactory {
```

Comment:
Without commenting on whether shared informers should offer this feature or not, I do think these options should be converted to a configuration struct (or use the builder pattern, but that seems unnecessarily complex for this). That way, additions won't break the function signature.